src/wallet/wallet.cpp

```diff
@@ -4502,34 +4484,40 @@ bool CWallet::IsLockedCoin(uint256 hash, unsigned int n) const
     return (setLockedCoins.count(outpt) > 0);
 }
 
-void CWallet::ListLockedCoins(std::vector<COutPoint>& vOutpts) const
+std::vector<COutPoint> CWallet::ListLockedCoins() const
 {
     AssertLockHeld(cs_wallet);
-    for (std::set<COutPoint>::iterator it = setLockedCoins.begin();
-         it != setLockedCoins.end(); it++) {
-        COutPoint outpt = (*it);
-        vOutpts.push_back(outpt);
+    std::vector<COutPoint> ret;
```

Comment:
add a reserve here for setLockedCoins.size?

Reply:
Or just return the set instead of a vet?

Reply:
Refactored `ListLockedCoins()`, keeping it as `std::vector` as it's exposed through interface and there are performance considerations for such use ([source](https://github.com/dashpay/dash/pull/6536#discussion_r1930069663)) but have tried to reduce calls to `ListLockedCoins()` to compensate for copying overhead. Resolved in latest push.